Amplify/Categories/DataStore/Model/Lazy/LazyReference.swift

```diff
@@ -0,0 +1,157 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Combine
+
+/// A Codable struct to hold key value pairs representing the identifier's field name and value.
+/// Useful for maintaining order for key-value pairs when used as an Array type.
+public struct LazyReferenceIdentifier: Codable {
```

Comment:
Is this custom key/value pair required? Is `[String: String]` not compatible with `Codable`?

Reply:
I don't believe the swift dictionary of String is codable, but I can check again to see if there are alternatives to this implementation

Reply:
ah, the reason was the need for an ordered dictionary, that's why it's array of codable structs right now. We are not doing key based lookups anywhere since 
- datastore model provider extracts a single item from the array and uses that as the identifier
- api model provider requires the ordered list of identifiers, it translates it to the GraphQL request in `ModelIdDecorator`